### PR TITLE
add provider ZALO - https://developers.zalo.me

### PR DESCRIPTION
**Pull request by tmtung144**
Zalo provider. Full docs [https://developers.zalo.me](https://developers.zalo.me)

### CODE REVIEW

**Comment by atymic** [comment]
@tmtung144 Please add documentation and link here.

**Comment by tmtung144** [comment]
> Vui lòng mở và liên kết đến một PR trên kho Tài liệu.

Please see [SocialiteProviders/Documentation/pull/88](https://github.com/SocialiteProviders/Documentation/pull/88)

File: src/Zalo/Provider.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id' => $user['id'],
+            'nickname' => null,
+            'name' => $user['name'],
+            // Emails are not officially supported in Zalo API
+            // This is a hypothetical email identified with the type "ID@zalo.me"
+            'email' => $user['id'] . '@zalo.me',
```
Comment (atymic) [inline src/Zalo/Provider.php:82]: can just remove this, it's not a required field

Comment (tmtung144) [reply src/Zalo/Provider.php]: check [SocialiteProviders/Providers/pull/371](https://github.com/SocialiteProviders/Providers/pull/371)

tks!

**Pull request merged**